### PR TITLE
Fixed sync between record value context selector and record store

### PR DESCRIPTION
**Pull request by lucasbordeau**
This PR introduces many improvements over the new profiling story feature, with new tests and some refactor with main : 
- Added use-context-selector for getting value faster in display fields and created useRecordFieldValue() hook and RecordValueSetterEffect to synchronize states
- Added performance test command in CI
- Refactored ExpandableList drill-downs with FieldFocusContext
- Refactored field button icon logic into getFieldButtonIcon util
- Added RelationFieldDisplay perf story
- Added RecordTableCell perf story
- First split test of useField.. hook with useRelationFieldDisplay()
- Fixed problem with set cell soft focus
- Isolated logic between display / soft focus and edit mode in the related components to optimize performances for display mode.
- Added warmupRound config for performance story decorator
- Added variance in test reporting

### CODE REVIEW

File: packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemAsFieldDefinition.ts [hunk 1]
```diff
@@ -24,25 +27,31 @@ export const formatFieldMetadataItemAsFieldDefinition = ({
...
+    placeHolder: field.label,
+    relationType: parseFieldRelationType(field),
+    relationFieldMetadataId,
+    relationObjectMetadataNameSingular:
+      relationObjectMetadataItem?.nameSingular ?? '',
+    relationObjectMetadataNamePlural:
+      relationObjectMetadataItem?.namePlural ?? '',
+    objectMetadataNameSingular: objectMetadataItem.nameSingular ?? '',
+    options: field.options,
+  };
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemAsFieldDefinition.ts:41]: Nice one! @lucasbordeau

File: packages/twenty-front/src/modules/object-record/record-store/contexts/RecordFieldValueSelectorContext.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import { createContext, useContextSelector } from 'use-context-selector';
+
+export type RecordFiedValue = {
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/object-record/record-store/contexts/RecordFieldValueSelectorContext.tsx:4]: good catches this typos

**Review by charlesBochet — COMMENTED** [review]
Great work, I'm aligned on the high level approach and the code seems not too hard to understand (even if I haven't seen use-context-selector before). 

It's polluted by stuff already present on main and there are also a few typos!
Let's fix these, and we can merge!

File: packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/RelationFieldDisplay.tsx [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-
 export const RelationFieldDisplay = () => {
-  const { fieldValue, fieldDefinition, maxWidth } = useRelationField();
+  const { fieldValue, fieldDefinition, maxWidth } = useRelationFieldDisplay();
 
   if (
     !fieldValue ||
     !fieldDefinition?.metadata.relationObjectMetadataNameSingular
-  )
+  ) {
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/RelationFieldDisplay.tsx:11]: Thanks bot, 100% agree with you.

**Review by charlesBochet — APPROVED** [review]
Bravo !

**Pull request merged**